HttpProtocolException.md

```diff
@@ -0,0 +1,29 @@
+# Detect HTTP/2.0 and HTTP/3.0 protocol errors
+
+HTTP2/2 and HTTP/3 defines protocol-level error codes in [RFC 7540 section 7](https://www.rfc-editor.org/rfc/rfc7540#section-7) and [RFC 9114 section 8.1](https://www.rfc-editor.org/rfc/rfc9114.html#section-8.1) (not to be confused with HTTP status codes!). Although such low-level error information is unimportant for most `HttpClient` users, it might be helpful in advanced HTTP/2 or HTTP/3 scenarios, notably grpc-dotnet, where distinguishing protocol errors is vital to implement client retries.
```

Comment:
- `HTTP2/2` -> `HTTP/2`
- Should we link to grpc-dotnet?
- I feel like we should give examples of protocol-level error codes, when we say that these are not to be confused with status codes, so it would be more obvious to the reader: "e.g. HTTP/2's `REFUSED_STREAM (0x7)` and HTTP/3's `H3_EXCESSIVE_LOAD (0x0107)`" -- or whatever error codes are more useful